src/cycles/limited_length.jl

```diff
@@ -0,0 +1,53 @@
+"""
+    simplecycles_limited_length(g, n, ceiling=10^6)
+
+Compute and return at most `ceiling` cycles of length at most `n` of
+the given graph. Both directed and undirected graphs are supported.
+
+### Performance
+The number of cycles grows very fast with the number of vertices and
+the allowed length of the cycles. This function is intended for
+finding short cycles. If you want to find cycles of any length in a
+directed graph, `simplecycles` or `simplecycles_iter` may be more
+efficient.
+"""
+function simplecycles_limited_length(graph::AbstractGraph, n::Integer,
```

Comment:
I would change this  to 
```
function simplecycles_limited_length(graph::AbstractGraph{T}, n::Int, ceiling = 10^6) where {T}
    cycles = Vector{T}[]
    n < 1 && return cycles
    cycle = T[]
    @inbounds for v = vertices(graph)
        ...
```
then it should a) work with all vertex types that can be ordered and b)
will use less memory when the vertex type is some smaller integer (e.g Int16).

The `@inbounds` here might not be necessary as the compiler seems to be able to infer this automatically in some cases nowadays.

Reply:
Right, of course. My unfamiliarity with the LightGraphs types shows but this much I should have figured already.

Reply:
The convention in LightGraphs is to explicitly call `Vector{T}()` instead of `T[]`.